src/arithmetic.jl

```diff
@@ -165,37 +193,39 @@ for T in (:Taylor1, :HomogeneousPolynomial, :TaylorN)
     @eval begin
         *(a::Bool, b::$T) = *(convert(Int, a), b)
         *(a::$T, b::Bool) = b * a
-        function *(a::RealOrComplex, b::$T)
+        function *{T<:NumberNotSeries}(a::T, b::$T)
             @inbounds aux = a * b.coeffs[1]
             v = Array{typeof(aux)}(length(b.coeffs))
-            @simd for i in eachindex(v)
-                @inbounds v[i] = a * b.coeffs[i]
-            end
+            v .= a .* b.coeffs
             $T(v, b.order)
         end
-        *(b::$T, a::RealOrComplex) = a * b
-        if $T != Taylor1
-            function *{T<:RealOrComplex}(a::Taylor1{T}, b::$T{Taylor1{T}})
-                @inbounds aux = a * b.coeffs[1]
-                S = typeof(aux)
-                coeffs = Array{S}(length(b.coeffs))
-                @simd for i in eachindex(coeffs)
-                    @inbounds coeffs[i] = a * b.coeffs[i]
-                end
-                return $T(coeffs, b.order)
-            end
-            *{T<:RealOrComplex}(b::$T{Taylor1{T}}, a::Taylor1{T}) = a * b
-            function *{T<:RealOrComplex,R<:RealOrComplex}(a::Taylor1{T},
-                    b::$T{Taylor1{R}})
-                S = promote_type(T,R)
-                return convert(Taylor1{S}, a) * convert($T{Taylor1{S}}, b)
-            end
-            *{T<:RealOrComplex,R<:RealOrComplex}(b::$T{Taylor1{T}},
-                a::Taylor1{R}) = a * b
+        *{T<:NumberNotSeries}(b::$T, a::T) = a * b
+    end
+end
+
+for T in (:HomogeneousPolynomial, :TaylorN)
+    @eval begin
+        function *{T<:NumberNotSeries,S<:NumberNotSeries}(a::Taylor1{T}, b::$T{Taylor1{S}})
+            @inbounds aux = a * b.coeffs[1]
+            R = typeof(aux)
+            coeffs = Array{R}(length(b.coeffs))
+            coeffs .= a .* b.coeffs
```

Comment:
It feels like these three lines form a recurring pattern that should be factored out into a function somehow.